src/mame/heathkit/tlb.cpp

```diff
@@ -0,0 +1,751 @@
+	GFXDECODE_ENTRY(":tlb:chargen", 0x0000, h19_charlayout, 0, 1)
+GFXDECODE_END
+
+
+/* Input ports */
+static INPUT_PORTS_START( tlb )
+
+	PORT_START("MODIFIERS")
+	// bit 0 connects to B8 of MM5740 - low if either shift key is
+	// bit 7 is low if a key is pressed
+	PORT_BIT(0x002, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_NAME("CapsLock")   PORT_CODE(KEYCODE_CAPSLOCK) PORT_TOGGLE
+	PORT_BIT(0x004, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_NAME("Break")      PORT_CODE(KEYCODE_PAUSE)
+	PORT_BIT(0x008, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_NAME("OffLine")    PORT_CODE(KEYCODE_F12)       PORT_TOGGLE
+	PORT_BIT(0x010, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_NAME("CTRL")       PORT_CODE(KEYCODE_LCONTROL)  PORT_CHAR(UCHAR_MAMEKEY(LCONTROL))
+	PORT_BIT(0x020, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_NAME("LeftShift")  PORT_CODE(KEYCODE_LSHIFT)    PORT_CHAR(UCHAR_SHIFT_1)
+	PORT_BIT(0x040, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_NAME("Repeat")     PORT_CODE(KEYCODE_LALT)
+	PORT_BIT(0x100, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_NAME("RightShift") PORT_CODE(KEYCODE_RSHIFT)
+	PORT_BIT(0x200, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_NAME("Reset")      PORT_CODE(KEYCODE_F10)
+
+	PORT_START("X1")
+	PORT_BIT(0x001, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_NAME("/")          PORT_CODE(KEYCODE_SLASH)      PORT_CHAR('/') PORT_CHAR('?')
+	PORT_BIT(0x002, IP_ACTIVE_LOW, IPT_UNUSED)
+	PORT_BIT(0x004, IP_ACTIVE_LOW, IPT_UNUSED)
+	PORT_BIT(0x008, IP_ACTIVE_LOW, IPT_UNUSED)
+	PORT_BIT(0x010, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_NAME("k_X2")
+	PORT_BIT(0x020, IP_ACTIVE_LOW, IPT_UNUSED)
+	PORT_BIT(0x040, IP_ACTIVE_LOW, IPT_KEYBOARD)                         PORT_CODE(KEYCODE_0_PAD)      PORT_CHAR(UCHAR_MAMEKEY(0_PAD))
+	PORT_BIT(0x080, IP_ACTIVE_LOW, IPT_KEYBOARD)                         PORT_CODE(KEYCODE_DEL_PAD)    PORT_CHAR(UCHAR_MAMEKEY(DEL_PAD))
+	PORT_BIT(0x100, IP_ACTIVE_LOW, IPT_KEYBOARD)                         PORT_CODE(KEYCODE_ENTER_PAD)  PORT_CHAR(UCHAR_MAMEKEY(ENTER_PAD))
+	PORT_BIT(0x200, IP_ACTIVE_LOW, IPT_UNUSED)
+
+	PORT_START("X2")
+	PORT_BIT(0x001, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_NAME(";")          PORT_CODE(KEYCODE_COLON)      PORT_CHAR(';') PORT_CHAR(':')
+	PORT_BIT(0x002, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_NAME("\'")         PORT_CODE(KEYCODE_QUOTE)      PORT_CHAR('\'') PORT_CHAR('"')
+	PORT_BIT(0x004, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_NAME("{")          PORT_CODE(KEYCODE_CLOSEBRACE) PORT_CHAR('{') PORT_CHAR('}')
+	PORT_BIT(0x008, IP_ACTIVE_LOW, IPT_KEYBOARD) PORT_NAME("Return")     PORT_CODE(KEYCODE_ENTER)      PORT_CHAR(13)
+	PORT_BIT(0x010, IP_ACTIVE_LOW, IPT_UNUSED)
+	PORT_BIT(0x020, IP_ACTIVE_LOW, IPT_UNUSED)
+	PORT_BIT(0x040, IP_ACTIVE_LOW, IPT_KEYBOARD)                         PORT_CODE(KEYCODE_1_PAD)      PORT_CHAR(UCHAR_MAMEKEY(1_PAD))
+	PORT_BIT(0x080, IP_ACTIVE_LOW, IPT_KEYBOARD)                         PORT_CODE(KEYCODE_2_PAD)      PORT_CHAR(UCHAR_MAMEKEY(2_PAD))
+	PORT_BIT(0x100, IP_ACTIVE_LOW, IPT_KEYBOARD)                         PORT_CODE(KEYCODE_3_PAD)      PORT_CHAR(UCHAR_MAMEKEY(3_PAD))
```

Comment:
Should the keypad number keys have their alternate function noted in a `PORT_NAME`?  (For these three, I believe it’s IL, Down and DL.)

Reply:
Based on your previous comment - https://github.com/mamedev/mame/pull/10958#discussion_r1141049291 I thought you didn't want the PORT_NAME specified for these keys. I'll add them back and put the added info.